src/reactivity.ts

```diff
@@ -138,13 +156,54 @@ function unregisterObserver(observer: Observer) {
   observerAtoms.delete(observer);
 }
 
-export function useState(state: any): Atom {
+export function useState(state: any, observer?: Observer): Atom {
   if (!isTrackable(state)) {
     throw new Error("Argument is not trackable");
   }
   const node = getCurrent()!;
-  const observer = () => node.render();
-  const unregisterObserver = registerObserver(observer);
+  observer = () => node.render();
+  const unregisterObserver = registerObserver(observer, { parent: node.parent });
```

Comment:
Two observers should not be associated with same node!
Alternative: share observer for same node